GiniVision/Classes/Core/Multipage Review/MultipageReviewViewController.swift

```diff
@@ -415,29 +435,34 @@ extension MultipageReviewViewController {
     
     @objc fileprivate func deleteImageButtonAction() {
         if let currentIndexPath = visibleCell(in: self.mainCollection) {
-            let documentToDelete = documentRequests[currentIndexPath.row]
-            documentRequests.remove(at: currentIndexPath.row)
-            mainCollection.deleteItems(at: [currentIndexPath])
-            
-            pagesCollection.performBatchUpdates({
-                self.pagesCollection.deleteItems(at: [currentIndexPath])
-            }, completion: { [weak self] _ in
-                guard let `self` = self else { return }
-                if self.documentRequests.count > 0 {
-                    if currentIndexPath.row != self.documentRequests.count {
-                        var indexes = IndexPath.indexesBetween(currentIndexPath,
-                                                               and: IndexPath(row: self.documentRequests.count,
-                                                                              section: 0))
-                        indexes.append(currentIndexPath)
-                        self.pagesCollection.reloadItems(at: indexes)
-                    }
-                    
-                    self.selectItem(at: min(currentIndexPath.row, self.documentRequests.count - 1))
-                }
-                self.delegate?.multipageReview(self, didDelete: documentToDelete)
-            })
+            deleteItem(at: currentIndexPath)
         }
     }
+    
+    public func deleteItem(at indexPath: IndexPath, completion: (() -> Void)? = nil) {
+        let documentToDelete = documentRequests[indexPath.row]
+        documentRequests.remove(at: indexPath.row)
+        mainCollection.deleteItems(at: [indexPath])
+        
+        pagesCollection.performBatchUpdates({
+            self.pagesCollection.deleteItems(at: [indexPath])
+        }, completion: { [weak self] _ in
+            guard let `self` = self else { return }
+            if self.documentRequests.count > 0 {
+                if indexPath.row != self.documentRequests.count {
+                    var indexes = IndexPath.indexesBetween(indexPath,
+                                                           and: IndexPath(row: self.documentRequests.count,
+                                                                          section: 0))
+                    indexes.append(indexPath)
+                    self.pagesCollection.reloadItems(at: indexes)
+                }
+                
+                self.selectItem(at: min(indexPath.row, self.documentRequests.count - 1))
+            }
+            self.delegate?.multipageReview(self, didDelete: documentToDelete)
+            completion?()
```

Comment:
A completion handler, that calls a delegate method and calls another completion handler? It's almost too much!

Reply:
I changed the order of the animation, so I can get rid of the completion handler and make it more clear.